integrations/aws/main.py

```diff
@@ -0,0 +1,196 @@
+                            instance_definition.update(
+                                {
+                                    KIND_PROPERTY: kind,
+                                    ACCOUNT_ID_PROPERTY: account_id,
+                                    REGION_PROPERTY: region,
+                                    IDENTIFIER_PROPERTY: instance.id,
+                                }
+                            )
+                            yield fix_unserializable_date_properties(
+                                instance_definition
+                            )
+        except Exception as e:
+            logger.error(f"Failed to list EC2 Instance in region: {region}; error {e}")
+
+
+@ocean.router.post("/webhook")
+async def webhook(request: Request) -> dict[str, Any]:
+    validate_request(request)
+    try:
+        body = await request.json()
+        logger.info(f"Received AWS Webhook request body: {body}")
+        resource_type = body.get("resource_type")
+        identifier = body.get("identifier")
+        account_id = body.get("accountId")
+        region = body.get("awsRegion")
+
+        with logger.contextualize(
+            account_id=account_id, resource_type=resource_type, identifier=identifier
+        ):
+            if not resource_type or not identifier or not account_id:
+                raise ValueError(
+                    "Resource type, Identifier or Account id was not found in webhook body"
+                )
+
+            matching_resource_configs = get_matching_kinds_from_config(resource_type)
+
+            if not matching_resource_configs:
+                raise ValueError(
+                    "Resource type not found in port app config, update port app config to include the resource type"
+                )
+
+            logger.debug("Querying full resource")
+            resource = await describe_single_resource(
+                resource_type, identifier, account_id, region
+            )
+            for resource_config in matching_resource_configs:
+                blueprint = resource_config.port.entity.mappings.blueprint.strip('"')
+                if not resource:  # Resource probably deleted
+                    logger.info("Resource not found in AWS, un-registering from port")
+                    await ocean.unregister(
+                        [
+                            Entity(
+                                blueprint=blueprint,
+                                identifier=identifier,
+                            )
+                        ]
+                    )
+                    continue
+
+                logger.info("Resource found in AWS, registering change in port")
+                resource.update(
+                    {
+                        KIND_PROPERTY: resource_type,
+                        ACCOUNT_ID_PROPERTY: account_id,
+                        REGION_PROPERTY: region,
+                        IDENTIFIER_PROPERTY: identifier,
+                    }
+                )
```

Comment:
please remove anything that can  already be found in the playload

Reply:
resources that are not using `cloudcontrol` do not always use `Identifier` as their identifier key, for example
in EC2 case it is listed as `InstanceId`